### PR TITLE
[Rubocop] 0.50.0 update

### PR DESCRIPTION
**Pull request by JelF**
- **moved rspec config to separate file**

- **updated rubocop to 0.50.0** (https://github.com/bbatsov/rubocop/blob/master/CHANGELOG.md#0500-2017-09-14)

- Lint/DuplicateMethods now uderstands attr_* methods and aliases.
  **Mass violations found** ☹

- new cop Style/RedundantConditional is good.
  And we have no violaions!

- new cop Rails/HasManyOrHasOneDependent is scary
  Disabled it globaly, may be we can enable it in some projects

- new cop Style/Dir is good.
  `__dir__` seems better than `File.expand_path(File.dirname(__FILE__))`

- **new cop Lint/RescueWithoutErrorClass** could be good.
  However, we forced `rescue` instead of `rescue StandardError`.
  Disabled it for now, but maybe we will force it instead?

- new cop Performance/UnfreezeString forces `+""`
  instead of `String.new`. It's cool

- **updated rubocop-rspec to 1.18.0** (https://github.com/backus/rubocop-rspec/blob/master/CHANGELOG.md#1180-2017-09-29)

- new cop FactoryGirl/DynamicAttributeDefinedStatically is good.
  Mass violations in code! Real shit because all of them are bugs.
  Enabled it

- new cops RSpec/Align{Left,Right}LetBrace force antipattern.
  Disabled them (which is default)

- **new cop RSpec/LetBeforeExamples** is good.
  I consider it quite useful, but we may need a discussion.
  Enabled it

- new cop RSpec/MultipleSubjects is good.
  Enabled it

- new cop RSpec/ReturnFromStub is good.
  We have mass violations, but it should not be a problem. It's only
  about style. Enabled it

- new cop RSpec/VoidExpect is good.
  We had violations in past and they always were errors. Enabled it

- new cop RSpec/PredicateMatcher is damned.
  Predicate matchers' error messages are shitty, we should not use them

- new cop RSpec/InvalidPredicateMatcher.
  Haven't found any offences so enabled it. No idea what does it do

- new cop RSpec/ExpectInHook is good.
  But it is too good for us. Disabled

- **new cop Capybara/FeatureMethods** is strange.
  Disabled it beacause `scenario` and `it` means different things

- new cop Capybara/CurrentPathExpectation.
  They should have a reason to make it.
  Enabled it due to zero violations in our code

### CODE REVIEW

File: rubocop/lib/rubocop.yml [hunk 1]
```diff
@@ -247,23 +221,20 @@ Style/PerlBackrefs:
   # Disable localy if realy neeeded
   Enabled: true
 
-Style/PredicateName:
```
Comment (tycooon) [inline rubocop/lib/rubocop.yml:250]: Why is this removed?

Comment (JelF) [reply rubocop/lib/rubocop.yml:250]: Renamed to Naming/PredicateName

File: rubocop/lib/rubocop.yml [hunk 2]
```diff
@@ -165,10 +140,6 @@ Security/YAMLLoad:
 
 # Style
 
-Style/AccessorMethodName:
```
Comment (JelF) [inline rubocop/lib/rubocop.yml:168]: Renamed to Naming/AccessorMethodName

**Review by tycooon — APPROVED** [review]
